src/App.tsx

```diff
@@ -1,24 +1,234 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
+import cn from 'classnames';
 import { UserWarning } from './UserWarning';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList';
+import { ErrorMessage } from './types/ErrorMessage';
+import {
+  createTodo,
+  deleteTodo,
+  getTodos,
+  updateTodo,
+} from './api/todos';
+import { Filter } from './types/Fillter';
+import { TodosFilter } from './components/TodosFillter';
 
-const USER_ID = 0;
+const USER_ID = 11926;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isError, setIsError] = useState<boolean>(false);
+  const [error, setError] = useState<ErrorMessage | null>(null);
+  const [todoTitle, setTodoTitle] = useState<string>('');
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [filterBy, setFilterBy] = useState<Filter>(Filter.all);
+  const [processingIds, setProcessingIds] = useState<number[]>([]);
+
+  const showError = (errorType: ErrorMessage) => {
+    setIsError(true);
+    setError(errorType);
+    setTimeout(() => {
+      setIsError(false);
+      setError(null);
+    }, 3000);
+  };
+
+  const filteredTodos = useMemo(() => {
+    switch (filterBy) {
+      case Filter.active:
+        return todos.filter(todo => !todo.completed);
+      case Filter.completed:
+        return todos.filter(todo => todo.completed);
+      default:
+        return todos;
+    }
+  }, [filterBy, todos]);
+
+  const isAllCompleted
+    = todos.filter(todo => todo.completed).length === todos.length;
+
+  const isAllUncompleted
+    = todos.filter(todo => !todo.completed).length === todos.length;
+
+  const resetField = (): void => {
+    setTodoTitle('');
+    setIsError(false);
+    setError(null);
+  };
+
+  const handleDeleteTodo = (todoId: number) => {
+    setProcessingIds(current => [...current, todoId]);
+    deleteTodo(todoId)
+      .then(() => setTodos(
+        currentTodos => currentTodos.filter(todo => todo.id !== todoId),
+      ))
+      .catch(() => {
+        showError(ErrorMessage.deleteError);
+      })
+      .finally(() => setProcessingIds([]));
+  };
+
+  const handleUpdateTodo = (todo: Todo) => {
+    setProcessingIds(current => [...current, todo.id]);
+    const currentTodo = todos.filter(toDo => toDo.id === todo.id)[0];
```

Comment:
you use `filter` like `find`.
```suggestion
    const currentTodo = todos.find(toDo => toDo.id === todo.id);
```

Reply:
or even `findIndex`